services/webnn/ort/graph_builder_ort.cc

```diff
@@ -945,9 +1100,19 @@ void GraphBuilderOrt::AddElementWiseLogicalOperation(
       if (!bool_operands_.count(element_wise_binary->lhs_operand_id)) {
         lhs = PrependCast(lhs, ONNX_TENSOR_ELEMENT_DATA_TYPE_BOOL);
       }
+      // If the current operand is inserted a cast, find the bool operand
+      // through the original operand.
+      if (bool_to_uint8_operands_.count(element_wise_binary->lhs_operand_id)) {
```

Comment:
Make it else, because bool_operands_ should have this input operand for this case:
```suggestion
      else if (bool_to_uint8_operands_.count(element_wise_binary->lhs_operand_id)) {
```

Reply:
Done.